packages/models-library/src/models_library/projects_nodes_io.py

```diff
@@ -73,17 +94,13 @@ class Config:
 class BaseFileLink(BaseModel):
     """Base class for I/O port types with links to storage services"""
 
-    # TODO: constructor will always cast to str here. We should perhaps set is as str. Actually
-    # if we want to do hash in inputs/outputs ... we should have a single type for identifiers
-    # Recall lru_cache options regarding types!!
-    store: Union[str, int] = Field(
+    store: LocationID = Field(
```

Comment:
TIP: check https://pydantic-docs.helpmanual.io/usage/types/#discriminated-unions-aka-tagged-unions

Reply:
yep. I think that is a very good solution